models/balancer/arbitrum/balancer_v2_arbitrum_pools_tokens_weights.sql

```diff
@@ -0,0 +1,51 @@
+{{
+    config(
+        schema = 'balancer_v2_arbitrum',
+        alias='pools_tokens_weights',
+        post_hook='{{ expose_spells(\'["arbitrum"]\',
+                                    "project",
+                                    "balancer",
+                                    \'["metacrypto", "jacektrocinski"]\') }}'
+    ) 
+}}
+
+--
+-- Balancer v2 Pools Tokens Weights
+--
+SELECT
+    registered.`poolId` AS pool_id,
+    tokens.token_address,
+    weights.normalized_weight / POWER(10, 18) AS normalized_weight
+FROM {{ source('balancer_v2_arbitrum', 'Vault_evt_PoolRegistered') }} registered
+INNER JOIN {{ source('balancer_v2_arbitrum', 'WeightedPoolFactory_call_create') }} call_create
+    ON call_create.call_tx_hash = registered.evt_tx_hash
+    LATERAL VIEW posexplode(call_create.tokens) tokens AS pos, token_address
+    LATERAL VIEW posexplode(call_create.weights) weights AS pos, normalized_weight
+WHERE tokens.pos = weights.pos
+
```

Comment:
The `project_start_date` for this model is `2021-08-26`.

However since all parts in `union all` using inner join to `{{ source('balancer_v2_arbitrum', 'Vault_evt_PoolRegistered') }}`, we can simply add an incremental filter here.
```suggestion
WHERE tokens.pos = weights.pos
    {% if is_incremental() %}
    AND registered.evt_block_time >= date_trunc("day", now() - interval '1 week')
    {% endif %}
```